source/vir/src/recursion.rs

```diff
@@ -522,6 +524,8 @@ pub(crate) fn expand_call_graph(
                                 // f1 --> D.f2
                                 Some(&method_map[&(x.clone(), datatype.clone())])
                             }
```

Comment:
When `method_map` is changed to account for all types, not just `Datatype`s, then the `TypX::Datatype` case can be generalized so that it includes `Lambda`.